core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -485,7 +520,18 @@ class Partition(val topic: String,
     laggingReplicas
   }
 
-  def appendRecordsToLeader(records: MemoryRecords, isFromClient: Boolean, requiredAcks: Int = 0) = {
+  def appendRecordsToFollower(records: MemoryRecords, isFuture: Boolean) {
```

Comment:
Previously I had a method called `appendRecordsToFutureReplica`. I merged it with `appendRecordsToFollower` to reduce the number of methods. Since this is misleading, I think it may be better to add method `appendRecordsToFutureReplica`, which is more intuitive and more consistent with the name `appendRecordsToLeader`. What do you think?